include/Gaffer/ThreadMonitor.h

```diff
@@ -42,6 +42,21 @@
 
 #include <unordered_map>
 
+namespace std
+{
+
+template <>
+struct hash<Gaffer::ConstPlugPtr>
+{
+	size_t operator()( const Gaffer::ConstPlugPtr &p ) const
+	{
+		return std::hash<const void*>()( p.get() );
+	}
+};
+
+} // namespace std
```

Comment:
Why is this necessary? Is it because Image Engine is using some old Boost where `std::hash` hasn't been specialised? If so, at a minimum can we comment this and `#ifdef` it so we're not defining a conflicting specialisation for everyone using the right boost version?